pkg/utils/release/controller.go

```diff
@@ -121,17 +108,120 @@ func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1a
 	return release, err
 }
 
-// Get ReleaseLink object from a given namespace
-func (s *SuiteController) GetReleaseLink(name string, namespace string) (*v1alpha1.ReleaseLink, error) {
-	namespacedName := types.NamespacedName{
+// GetReleasePlanAdmission returns the ReleasePlanAdmission with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlanAdmission(name, namespace string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	releasePlanAdmission := &appstudiov1alpha1.ReleasePlanAdmission{}
+
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, releasePlanAdmission)
+
+	return releasePlanAdmission, err
+}
+
+// DeleteReleasePlanAdmission deletes the ReleasePlanAdmission resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlanAdmission has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlanAdmission(name, namespace string, failOnNotFound bool) error {
+	releasePlanAdmission := appstudiov1alpha1.ReleasePlanAdmission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	err := s.KubeRest().Delete(context.TODO(), &releasePlanAdmission)
+	if err != nil && !failOnNotFound && k8sErrors.IsNotFound(err) {
+		err = nil
+	}
+	return err
+}
+
+// CreateReleasePlan creates a new ReleasePlan using the given parameters.
+func (s *SuiteController) CreateReleasePlan(name, namespace, application, targetNamespace, autoReleaseLabel string) (*appstudiov1alpha1.ReleasePlan, error) {
+	var releasePlan *appstudiov1alpha1.ReleasePlan = &appstudiov1alpha1.ReleasePlan{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+			Name:         name,
+			Namespace:    namespace,
+			Labels: map[string]string{
```

Comment:
This is overwritten by lines 157-161, so I think it is clearer to remove it here

Reply:
I've removed the lines 147-149, 
made a build and ran test , it failed with error of nil mapping : 
**assignment to entry in nil map**
I would keep it like this and maybe do improvement later on, if it's fine by you @johnbieren

Reply:
I think you can remove line 148 then so it is still declaring labels as a map, but with no entries

Reply:
missed this comment, 
will include this fix in other PR